frontend/public/components/row-filters.tsx

```diff
@@ -0,0 +1,348 @@
+            </span>
+            <span className="co-filter-dropdown-item__name">{item.title}</span>
+            <Badge key={item.id} isRead>
+              {_.countBy(data, grp.reducer)?.[item.id] ?? '0'}
+            </Badge>
+          </div>
+        </DropdownItem>
+      ))}
+    </DropdownGroup>
+  ));
+
+const getSelectedRowFilters = (rowFilters: RowFilter[]) =>
+  rowFilters.reduce((acc, curr) => [...acc, ...curr?.selected], []);
+
+const DropdownFilter_: React.FC<DropdownFilterProps> = (props) => {
+  const { rowFilters = [], data, showNameFilter = true, showLabelFilter = false } = props;
+
+  const [labelFilters, setLabelFilters] = React.useState<string[]>([]);
+  const [nameFilter, setNameFilter] = React.useState('');
+  const [labelFilterText, setLabelFilterText] = React.useState('');
+  const [filterType, setFilterType] = React.useState(FilterType.NAME);
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState(() => getSelectedRowFilters(rowFilters));
+
+  const filters: Filter = React.useMemo(
+    () =>
+      rowFilters.reduce((acc, curr) => {
+        const items = _.map(curr.items, 'id');
+        acc[curr.filterGroupName] = items;
+        return acc;
+      }, {}),
+    [rowFilters],
+  );
+
+  // { 'Xstatus' = 'x-y-status' ...}
+  const filterKeys: FilterKeys = React.useMemo(
+    () =>
+      rowFilters.reduce((acc, curr) => {
+        const str = `${storagePrefix}${curr.type}`;
+        acc[curr.filterGroupName] = str;
+        return acc;
+      }, {}),
+    [rowFilters],
+  );
+
+  // Parse state from URL
+  React.useEffect(() => {
+    let q: string = '',
+      name: string;
+    const rowFiltersFromURL: string[] = [];
+
+    if (window.location.search) {
+      const params = new URLSearchParams(window.location.search);
+      q = params.get('q');
+      name = params.get('name');
+      _.map(_.values(filterKeys), (f) => {
+        const vals = params.get(f);
+        if (vals) {
+          rowFiltersFromURL.push(...vals.split(','));
+        }
+      });
+    }
+    const labels = q ? q.split(',') : [];
+    labels.length > 0 && setLabelFilters(labels);
+    setNameFilter(name || '');
+    setSelected([...selected, ...rowFiltersFromURL]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  /* Logic for Name and Label Filter */
+
+  const applyFilter = (value: string | string[], type: FilterType) => {
+    const filter = type === FilterType.NAME ? value : { all: value };
+    props.reduxIDs.forEach((id) => props.filterList(id, filterTypeMap[type], filter));
+  };
+
+  const addLabelFilter = (filter: string) => {
+    setLabelFilters([...labelFilters, filter]);
+    setQueryArgument('q', [...labelFilters, filter].join(','));
+    applyFilter([...labelFilters, filter], FilterType.LABEL);
+    setLabelFilterText('');
+  };
+
+  const removeLabelFilter = (filter: string) => {
+    const filtered = labelFilters.filter((l) => l !== filter);
```

Comment:
I would store the filters interally as a `Set`, then you can just remove the value from the `Set`.